File: app/src/main/java/com/bitflaker/lucidsourcekit/notification/NotificationScheduleData.java
package com.bitflaker.lucidsourcekit.notification;

import com.bitflaker.lucidsourcekit.general.Tools;

import java.util.Calendar;

public class NotificationScheduleData {
    private String id;
    private long timestamp;
    private boolean isNextDay;

    public NotificationScheduleData(long timestamp, String id) {
        this.id = id;
        this.timestamp = timestamp;
        this.isNextDay = false;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public boolean isNextDay() {
        return isNextDay;
    }

    public void setNextDay(boolean nextDay) {
        isNextDay = nextDay;
    }

    public long getScheduleTime() {
        Calendar cal = Calendar.getInstance();
        cal.setTimeInMillis(Tools.getMidnightTime() + getTimestamp());
        if(isNextDay) {
            cal.add(Calendar.HOUR_OF_DAY, 24);
        }
        return cal.getTimeInMillis();
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/main/notification/NotificationScheduleData.kt
package com.bitflaker.lucidsourcekit.main.notification

import com.bitflaker.lucidsourcekit.utils.Tools
import java.util.Calendar

class NotificationScheduleData(var timestamp: Long, var id: String?) {
    var isNextDay: Boolean = false

    val scheduleTime: Long
        get() {
            val cal = Calendar.getInstance()
            cal.timeInMillis = Tools.getMidnightTime() + timestamp
            if (isNextDay) {
                cal.add(Calendar.HOUR_OF_DAY, 24)
            }
            return cal.getTimeInMillis()
        }
}
